src/Symfony/Component/ObjectMapper/Metadata/ReverseClassObjectMapperMetadataFactory.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Metadata;
+
+use Symfony\Component\ObjectMapper\Attribute\Map;
+
+/**
+ * @author Florent Blaison <florent.blaison@gmail.com>
+ */
+final class ReverseClassObjectMapperMetadataFactory implements ObjectMapperMetadataFactoryInterface
+{
+    private array $attributesCache = [];
+
+    public function __construct(
+        private readonly ObjectMapperMetadataFactoryInterface $objectMapperMetadataFactory,
+        private readonly array $classMap,
+    ) {
+    }
+
+    public function create(object $object, ?string $property = null, array $context = []): array
+    {
+        $class = $object::class;
+        $key = $class.($property ?? '');
```

Comment:
It might be great to add a separator like `.` to avoid unexpected collisions

Reply:
Class `Foo` with a `barbaz` property and class `Foobar` with a `baz` property.